OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -96,38 +136,38 @@ public override Activity Tick(Actor self)
 			if (useLastVisibleTarget && !lastVisibleTarget.IsValidFor(self))
 				return NextActivity;
 
-			var pos = self.CenterPosition;
 			var checkTarget = useLastVisibleTarget ? lastVisibleTarget : target;
-
-			if (!soundPlayed && aircraft.Info.TakeoffSounds.Length > 0 && self.IsAtGroundLevel())
-			{
-				Game.Sound.Play(SoundType.World, aircraft.Info.TakeoffSounds, self.World, aircraft.CenterPosition);
-				soundPlayed = true;
-			}
-
-			// We are taking off, so remove influence in ground cells.
-			if (self.IsAtGroundLevel())
-				aircraft.RemoveInfluence();
+			var delta = checkTarget.CenterPosition - self.CenterPosition;
+			var desiredFacing = !aircraft.Info.CanHover || delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : aircraft.Facing;
```

Comment:
Did you mean `var desiredFacing = aircraft.Info.CanHover && delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : aircraft.Facing;`?

The current line will always return `delta.Yaw.Facing` for non-hovering aircraft, even if `delta` is zero, meaning the yaw is bogus.

Reply:
I meant it the way I wrote it, because I took that from the old
https://github.com/OpenRA/OpenRA/pull/16584/commits/78c930c94b0e4709dbde2a5b16f3644d6ab1fb57#diff-558171f0691cfeefaa697d6b9dd0734bL126

I assume `var desiredFacing = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : aircraft.Facing;` would be the correct fix then.